### PR TITLE
add cli support

### PR DESCRIPTION
**Pull request by Ethan-Arrowood**
#### Checklist

- [ ] run `npm run test` and `npm run benchmark`
- [x] tests and/or benchmarks are included
- [x] documentation is changed or added
- [x] commit message and code follows [Code of conduct](https://github.com/fastify/fastify/blob/master/CODE_OF_CONDUCT.md)

### CODE REVIEW

**Comment by Ethan-Arrowood** [comment]
@fastify/fastify can I get a review on this new feature please and thank you? 

Additionally, I tried to write a unit test for it, but really struggled. My understanding of my problem was that `execSync` would return before the file operations completed and thus would break the tests. If anyone has an idea on how to properly test `cli.ts` let me know! You can check out the test I had written [here](https://github.com/fastify/compile-schemas-to-typescript/pull/6/commits/cfa85173f4abbdc115cfeab21fcbce66b6438202#diff-583d616f4b01bf1fa153e71c0cef7de7)

File: src/cli.ts [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+
+import compileSchemas from './index'
+
+async function cli () {
+  const [ input, output ] = process.argv.slice(2)
+  try {
+    await compileSchemas(input, output)
+  } catch (err) {
+    console.error(err)
+    process.exit(0)
```
Comment (Eomm) [inline src/cli.ts]:
```suggestion
    process.exit(1)
```

File: README.md [hunk 2]
```diff
@@ -2,6 +2,14 @@
 
 This module is a wrapper around the `json-schema-to-typescript` library that enables transformation of directories of schemas instead of individual schemas. If directory support is added to the `json-schema-to-typescript` library this module will be archived.
 
+## CLI
+
+This module works as an CLI. Install as a dependency and call `compile-schemas-to-typescript` from a npm script, or use `npx`
+
+```bash
+npx compile-schemas-to-typescript schema-dir types-output-dir
```
Comment (Eomm) [inline README.md]:
```suggestion
npx compile-schemas-to-typescript <schema-dir> <types-output-dir>
```

**Review by Eomm — COMMENTED** [review]
For the test, could not execute the cli with spawn and then, when it returns, check if it has created the files?

I use this approach for this cli:
https://github.com/Eomm/keepy-cli/blob/master/test/helper.js#L95

**Comment by Ethan-Arrowood** [comment]
Will it close before or after file operations are complete? I was using `exec` from `child-process` and it would return/resolve before the file writes completed

**Pull request merged**